samples/nextjs/scripts/templates/component-factory.ts

```diff
@@ -16,22 +20,71 @@ function generateComponentFactory(components: ComponentFile[]): string {
   return `/* eslint-disable */
 // Do not edit this file, it is auto-generated at build time!
 // See scripts/generate-component-factory.ts to modify the generation of this file.
+
+import dynamic from 'next/dynamic'
+
 ${components
-  .map((component) => `import * as ${component.moduleName} from '${component.path}';`)
+  .map((component) => {
+    if (isLazyLoadingModule(component.path)) {
+      const moduleName = removeModuleNameEnding(component.moduleName);
+      return `const ${moduleName} = {
+  module: () => import('${component.path}'),
+  element: () => dynamic(${moduleName}.module)
```

Comment:
Do we also want to give the ability to pass in a custom loading component, or set SSR to true?

nextjs docs: https://nextjs.org/docs/advanced-features/dynamic-import#with-custom-loading-component

Reply:
@CobyPear `ssr: true` by default, and it should be enabled.

Reply:
@CobyPear For now user will not have ability to pass LoadingComponent, because `componentFactory` called in `Placeholder` component. User can manually add Loading component if he will need it

Reply:
@illiakovalenko OK, works for me. Will resolve this and approve.